internal/layers/layers.go

```diff
@@ -154,7 +154,19 @@ func MountContainerLayers(ctx context.Context, layerFolders []string, guestRoot
 	if err != nil {
 		return "", fmt.Errorf("failed to add SCSI scratch VHD: %s", err)
 	}
-	containerScratchPathInUVM = scsiMount.UVMPath
+
+	// This handles the case where we want to share a containers scratch disk instead of mounting a
+	// new one. If the refcount is > 1 this means we've already mounted the scratch vhdx to the UVM.
+	// In OpenGCS they take the `ScratchPath` value in the CombinedLayers request and simply setup the
+	// upper and work directories for the overlay here. As we'd be passing in the exact same scratch
+	// space location this would run into trouble when going to setup a new overlay here as one already
+	// exists so we need to pass in a different value for `ScratchPath` that is still inside the mount point.
```

Comment:
This kind of seems like overkill to me. I think the below would be fine: 

```
        // This handles the case where we want to share a scratch disk for multiple containers instead 
        // of mounting a new one. Pass a unique value for `ScratchPath` to avoid container upper and 
        // work directories colliding in the UVM. 
```

Reply:
Sure

Reply:
Done